File: String Subsequence Game - GFG/string-subsequence-game.cpp
// { Driver Code Starts
// Initial Template for C++
#include <bits/stdc++.h>
using namespace std;

 // } Driver Code Ends
// User function Template for C++
class Solution {
  public:
    set<string> allPossibleSubsequences(string S) {
        // code here
        func("",0,S.length(),S);
       set<string> ans;
       for(int i=0;i<req.size();i++){
           if((req[i][0]=='a' || req[i][0]=='e' || req[i][0]=='i' || req[i][0]=='o'||req[i][0]=='u')  )
           {
               int n=req[i].size();
               if(req[i][n-1]!='a' and req[i][n-1]!='e' and req[i][n-1]!='i' and req[i][n-1]!='o' and req[i][n-1]!='u')
                   ans.insert(req[i]);
           }
       }
       return ans;
        
        
        
    }
    
    vector<string> req;
 void func(string t , int i , int n , string s){
     if(i==n){
        req.push_back(t);
        return ;
     }
  
   func(t,i+1,n,s);
   t+=s[i];
   func(t,i+1,n,s);
   
 }
    
    
    
};

// { Driver Code Starts.
int main() {
    int t;
    cin >> t;
    while (t--) {
        string S;
        cin >> S;
        Solution ob;
        set<string> ans = ob.allPossibleSubsequences(S);
        if (ans.size() == 0)
            cout << -1 << endl;
        else {
            for (auto i : ans) cout << i << " ";
            cout << endl;
        }
    }
    return 0;
}
  // } Driver Code Ends

File: String Subsequence Game - GFG/string-subsequence-game.java
// { Driver Code Starts
// Initial template for Java
import java.util.*;
import java.io.*;
class GFG {
    public static void main(String args[]) throws IOException {
        BufferedReader read =
            new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(read.readLine());
        while (t-- > 0) {
            String S = read.readLine();
            Solution ob = new Solution();
            TreeSet<String> ans = new TreeSet<String>();
            ans = ob.allPossibleSubsequences(S);
            Iterator it = ans.iterator();
            if (ans.size() == 0) {
                System.out.print(-1);
            } else
                while (it.hasNext()) {
                    System.out.print(it.next() + " ");
                }
            System.out.println();
        }
    }
}// } Driver Code Ends


// User function template for Java
class Solution
{
static TreeSet<String> allPossibleSubsequences(String s) {
List<Integer> list=new LinkedList<>();
TreeSet<String> ts=new TreeSet<>();
int ep,i,j,num,t,first,last;
String vowelString="aeiouAEIOU";
t=s.length()-1;
ep=(int)Math.pow(2,s.length())-1;
for(num=1;num<=ep;num++)
{
for(i=s.length()-1;i>=0;i--)
{
j=num>>i;
if((j&1)==1) list.add(t-i);
}
if(list.size()>1)
{
first=list.get(0);
last=list.get(list.size()-1);
if(vowelString.indexOf(s.charAt(first))!=-1 && vowelString.indexOf(s.charAt(last))==-1) 
{
String r="";
for(int x:list)
{
r=r+s.charAt(x);
}
ts.add(r);
}
}
list.clear();
}
return ts;
    }
}